### DISCUSSION TITLE
Building openssl for hp nonstop guardian environment.

### DISCUSSION THREAD

**Discussion by prasant-patra** [openssl/openssl#23108] [Q&A]
Hello all,

Recently we are trying to upgrade one of our application code to use openssl 3.0 in HP Nonstop OS platform.
Thought we can easily use openssl library available at https://ituglib.connect-community.org/apps/Ituglib site.
But that seems to give us some linking issue because it was built for OSS environment in Nonstop.
Our application runs on **GUARDIAN** environment, Hence we have to build openssl and then can link to our application.
Earlier we had built openssl 1.0.2 successfully for **GUARDIAN** from **OSS** environment.

In openssl documentation it has help file for nonstop **NOTES-NONSTOP.md**. Currently running configure script with below options from **OSS** terminal.
./Configure nonstop-nsx_g_tandem --prefix=/home/openssl/prasant/openssl_3.0_lib --openssldir=/home/openssl/prasant/openssl_3.0_lib/ssl no-shared no-threads no-makedepend --with-rand-seed=os
After that when I use make command on above generated Makefile, It gives below error. As we know kill api is undefined in guardian environment.

**Can someone tell me why it tries to compile non guardian code when we have set configure option **nonstop-nsx_g_tandem** ?
Am I using correct configure options to build for guardian environment?**
Appreciate your help to complete openssl build on nonstop platform!

c99: running /usr/cmplr/ccomfe --c99 -ZO2 -Zg3 -ZS/tmp/t738197510AABFHJ.T -ZtG -
ZR8 -m -Zr -ZT -D_TANDEM_SOURCE -ZNW203 -ZNW220 -ZNW272 -ZNW734 -ZNW770 -ZNW1506
 -Zz -I. -Iinclude -Iapps/include -I/usr/include -DOPENSSL_PIC -DOPENSSLDIR="/ho
me/openssl/prasant/openssl_3.0_lib/ssl" -DENGINESDIR="/home/openssl/prasant/open
ssl_3.0_lib/lib/engines-3" -DMODULESDIR="/home/openssl/prasant/openssl_3.0_lib/l
ib/ossl-modules" -DOPENSSL_BUILDING_OPENSSL -DOPENSSL_VPROC= -D_XOPEN_SOURCE -D_
XOPEN_SOURCE_EXTENDED=1 -D_TANDEM_SOURCE -DB_ENDIAN -DNO_GETPID -D_TANDEM_ARCH=3
 -DNDEBUG apps/lib/http_server.c
              (void)kill(kidpids[i], SIGTERM);
                    ^
"/home/openssl/prasant/openssl-3.0.12/apps/lib/http_server.c", line 97: error(11
4):
          identifier "kill" is undefined

1 error detected in the compilation of "apps/lib/http_server.c".
c99: /usr/cmplr/ccomfe exited, returning 2.
c99: Exiting with status 1; 0 error(s).

Thanks!
Prasant

**Comment by t8m** [comment]
Duplicate of #23107
